### PR TITLE
BREAKING CHANGE: rename database objects

### PR DESCRIPTION
**Pull request by andreaskoepf**
This should complete the work started by @alexandrelefourner to rename several database objects/models.

person -> user
work_package -> task
post -> message
thread -> message_tree

### CODE REVIEW

**Review by mjagkow — COMMENTED** [review]
Couple of issues, other looks good

File: backend/oasst_backend/exceptions.py [hunk 1]
```diff
@@ -27,21 +27,21 @@ class OasstErrorCode(IntEnum):
...
     # 2000-3000: prompt_repository
-    INVALID_POST_ID = 2000
-    POST_NOT_FOUND = 2001
+    INVALID_FRONTEND_MESSAGE_ID = 2000
+    MESSAGE_NOT_FOUND = 2001
     RATING_OUT_OF_RANGE = 2002
     INVALID_RANKING_VALUE = 2003
     INVALID_TASK_TYPE = 2004
     USER_NOT_SPECIFIED = 2005
     NO_THREADS_FOUND = 2006
```
[the remark this answers is not in the document]
Comment (andreaskoepf) [reply backend/oasst_backend/exceptions.py:36]: good catch

File: backend/main.py [hunk 2]
```diff
@@ -81,96 +81,97 @@ class DummyPost(pydantic.BaseModel):
...
                     ),
                 ]
 
-                for p in dummy_posts:
-                    wp = pr.fetch_workpackage_by_postid(p.task_post_id)
-                    if wp and not wp.ack:
+                for msg in dummy_messages:
+                    task = pr.fetch_task_by_frontend_message_id(msg.task_message_id)
+                    if task and not task.ack:
                         logger.warning("Deleting unacknowledged seed data work package")
```
[the remark this answers is not in the document]
Comment (andreaskoepf) [reply backend/main.py:146]: yip!

File: backend/oasst_backend/prompt_repository.py [hunk 3]
```diff
@@ -383,108 +385,110 @@ def store_text_labels(self, text_labels: protocol_schema.TextLabels) -> TextLabe
...
+        )
         if not replies:
             raise OasstError("No replies found", OasstErrorCode.NO_REPLIES_FOUND)
 
-        thread = self.fetch_thread(replies[0].thread_id)
-        thread = {p.id: p for p in thread}
-        thread_posts = [thread[replies[0].parent_id]]
+        message_tree = self.fetch_message_tree(replies[0].message_tree_id)
+        message_tree = {p.id: p for p in message_tree}
+        thread_messages = [message_tree[replies[0].parent_id]]
```
[the remark this answers is not in the document]
Comment (andreaskoepf) [reply backend/oasst_backend/prompt_repository.py]: ok, I renamed `thread_messages` -> `conversation`

**Comment by k-nearest-neighbor** [comment]
Have tested web up to this point / looked for anything to rename on the web side. No issues so far.

(Shows we have some work to do to hooking up the website to the api)

File: backend/main.py [hunk 4]
```diff
@@ -81,96 +81,97 @@ class DummyPost(pydantic.BaseModel):
...
-                        task_post_id="de111fa8",
-                        user_post_id="6f1d0711",
-                        parent_post_id=None,
+                dummy_messages = [
+                    DummyMessage(
+                        task_message_id="de111fa8",
+                        user_message_id="6f1d0711",
+                        parent_message_id=None,
                         text="Hi!",
                         role="user",
```
[the remark this answers is not in the document]
Comment (andreaskoepf) [reply backend/main.py]: old 'user' in context of conversations is now called 'prompter' (this includes protocol changes)

**Review by yk — APPROVED** [review]
huge props, thank you very much!

**Comment by andreaskoepf** [comment]
closes #124

**Pull request merged**